RELEASE_PROCESS.md

```diff
@@ -0,0 +1,98 @@
+# Release Process
+
+- [Release Process](#release-process)
+  - [Breaking Changes](#breaking-changes)
+  - [Release Procedure](#release-procedure)
+    - [Creating a new release branch](#creating-a-new-release-branch)
+    - [Cutting a new release](#cutting-a-new-release)
+    - [Tagging Procedure](#tagging-procedure)
+  - [Patch release Procedure](#patch-release-procedure)
+
+This document outlines the release process for Babylon node (babylond)
+
+Babylon follows [semantic versioning](https://semver.org), but with the following deviations to account for state-machine and API breaking changes.
+
+- State-machine breaking changes & API breaking changes will result in an increase of the minor version Y (0.Y.z).
+- All other changes will result in an increase of the patch version Z (0.y.Z).
+
+## Breaking Changes
+
+A change is considered to be ***state-machine breaking*** if it requires a coordinated upgrade for the network to preserve state compatibility
+Note that when bumping the dependencies of [Cosmos SDK](https://github.com/cosmos/cosmos-sdk) and [IBC](https://github.com/cosmos/ibc-go),
+we will only treat patch releases as non state-machine breaking.
+
+A change is considered to be ***API breaking*** if it modifies the provided API. This includes events, queries, CLI interfaces.
+
+## Release Procedure
+
+A _release_ is an increment of the second number (eg: `v0.1.0` → `v0.2.0`)
+
+**Note**: Generally, PRs should target either `main` or a long-lived feature branch (see [CONTRIBUTING.md](./CONTRIBUTING.md#pull-requests)).
+
+* Once the team feels that `main` is _**feature complete**_, we create a `release/v0.Y.x` branch (going forward known as release branch),
+  where `Y` is the minor version number, with patch part substituted to `x` (eg: v0.11.x).
+  * **PRs targeting directly a release branch can be merged _only_ when exceptional circumstances arise**.
```

Comment:
What happens on those exceptional situations? Let's say we find a bug that affects `release/v0.3.x` and `release/v0.4.x` (with that being the latest one).

We need to release new patches for v0.3.x and v0.4.x. How do we accomplish that?

Reply:
in general standard process would be to:
- create fix on main
- backport fix to `release/v0.3.x` and `release/v0.4.x` branches

Now if `release/v0.3.x` is old and we already diverged by a lot then:
- pr to main and back port to  `release/v0.4.x`
- direct pr with fix on `release/v0.3.x`